pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -252,6 +256,15 @@ type KeyManagementServiceSpec struct {
 	TokenSecretName string `json:"tokenSecretName,omitempty"`
 }
 
+type ServerSideEncryptionSpec struct {
+	// KMSEnabled enables SSE-KMS for the s3 protocol, please see https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingKMSEncryption.html
+	// +optional
+	KMSEnabled bool `json:"kmsEnabled,omitempty"`
```

Comment:
Perhaps I missed a discussion somewhere but why not have a string enum of the supported SSE instead of defining a property for each via a bool? I'm leaning towards something like `backend` perhaps and then we can pass `kms` or `s3`. Thoughts?

Reply:
I agree with Seb that this config layout is confusing. How do `KeyManagementService` at the level above and `KMSEnabled` interact? Why is `KMSEnabled` not assumed if `KeyManagementService` is specified?

Reply:
Originally it was a single string with an enum from [this discussion](https://github.com/rook/rook/pull/10204#pullrequestreview-961644319). Since combinations are allowed, it seemed better to split them up to separate settings, but perhaps there is a more clear way.

Reply:
@leseb @BlaineEXE @travisn : what about `serverSideEncryption` moving to `ObjectStoreSpec` since option introduced for RGW and rename it `awsServerSideEncryptionSpec`  which will be more clear and avoids this confusion

Reply:
Agreed it should move to the ObjectStoreSpec since currently it would apply to the ClusterSpec through the SecuritySpec, which doesn't make sense. 
Looking at the [aws sse doc](https://docs.aws.amazon.com/AmazonS3/latest/userguide/serv-side-encryption.html), it says the options for kms and s3 encryption are mutually exclusive. If that is the case, this could just be an enum string again, such as `awsServerSideEncryption: kms | s3`, right?